app/components/category-item.js

```diff
@@ -1,47 +1,121 @@
 import Ember from 'ember';
 
-export default Ember.Component.extend({
+const {
+  Component,
+  computed,
+  get,
+  inject: {
+    service,
+  },
+  set,
+} = Ember;
+
+
+/**
+ * `category-item` composes a category's icon, and description.
+ *
+ * ## default usage
+ *
+ * ```handlebars
+ * {{category-item category=category}}
+ * ```
+ *
+ * @class category-item
+ * @module Component
+ * @extends Ember.Component
+ */
+export default Component.extend({
   classNames: ['category-item'],
   classNameBindings: ['selected'],
   isLoading: false,
 
-  flashMessages: Ember.inject.service(),
-  userCategories: Ember.inject.service(),
+  /**
+   * @property flashMessages
+   * @type Ember.Service
+   */
+  flashMessages: service(),
+
+  /**
+   * @property userCategories
+   * @type Ember.Service
+   */
+  userCategories: service(),
 
-  selected: Ember.computed.notEmpty('userCategory'),
+  /**
+   * Returns if the category has been selected by the user.
+   *
+   * @property selected
+   * @type Boolean
+   */
+  selected: computed.notEmpty('userCategory'),
```

Comment:
Computed macros can be imported directly on one line like so:

import { notEmpty } from 'ember-computed';